src/Process/ExecutionLoop.php

```diff
@@ -245,4 +300,9 @@ protected function flushProcessOutput(ProcessWrapper $processWrapper): void
             $processWrapper->getClassName()
         );
     }
+
+    protected function parallelLimitReached(): bool
+    {
+        return count($this->processSet->get(ProcessWrapper::PROCESS_STATUS_PREPARED)) >= $this->parallelLimit;
```

Comment:
This is strange, isn`t it?

How current number of processes be greater than limit?

Reply:
Well, this is parallelism, who knows what could happen :rofl: . I prefer being a bit defensive there.